x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/AbstractMultivalueFunctionTestCase.java

```diff
@@ -123,7 +123,7 @@ protected static void bytesRefs(
         Function<DataType, DataType> expectedDataType,
         BiFunction<Integer, Stream<BytesRef>, Matcher<Object>> matcher
     ) {
-        for (DataType type : new DataType[] { DataType.KEYWORD, DataType.TEXT, DataType.IP, DataType.VERSION }) {
+        for (DataType type : new DataType[] { DataType.KEYWORD, DataType.TEXT, DataType.IP, DataType.VERSION, DataType.SEMANTIC_TEXT }) {
```

Comment:
For the mv functions, I only modified the tests that explicitly tested for keyword/text fields to also add tests for semantic_text. For most of the mv function tests I did not need to make any changes because I modified the `bytesRef` method here:

`mv_concat` and `mv_zip` do not use `bytesRef`, but use the `DataType.isString` method to only test string types, so implicitly they tests `semantic_text`:

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvConcatTests.java#L34-L44
https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvZipTests.java#L37-L44

the rest of the mv functions that take string params use `bytesRef` and did not need modifications:
https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvCountTests.java#L33

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvDedupeTests.java#L40
https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvFirstTests.java#L34
https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvLastTests.java#L34
https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvMaxTests.java#L35
https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvMinTests.java#L35